hathor/sysctl/p2p/manager.py

```diff
@@ -0,0 +1,143 @@
+# Copyright 2023 Hathor Labs
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from typing import List, Tuple
+
+from hathor.p2p.manager import ConnectionsManager
+from hathor.sysctl.exception import SysctlException
+from hathor.sysctl.sysctl import Sysctl
+
+
+def parse_text(text: str) -> List[str]:
+    """Parse text per line skipping empty lines and comments."""
+    ret: List[str] = []
+    for line in text.splitlines():
+        line = line.strip()
+        if not line:
+            continue
+        if line.startswith('#'):
+            continue
+        ret.append(line)
+    return ret
+
+
+class ConnectionsManagerSysctl(Sysctl):
+    def __init__(self, connections: ConnectionsManager) -> None:
+        super().__init__()
+
+        self.connections = connections
+        self.register(
+            'max_enabled_sync',
+            self.get_max_enabled_sync,
+            self.set_max_enabled_sync,
+        )
+        self.register(
+            'rate_limit.global.send_tips',
+            self.get_global_send_tips_rate_limit,
+            self.set_global_send_tips_rate_limit,
+        )
+        self.register(
+            'sync_update_interval',
+            self.get_lc_sync_update_interval,
+            self.set_lc_sync_update_interval,
+        )
+        self.register(
+            'force_sync_rotate',
+            None,
+            self.set_force_sync_rotate,
+        )
+        self.register(
+            'always_enable_sync',
+            self.get_always_enable_sync,
+            self.set_always_enable_sync,
+        )
+        self.register(
+            'always_enable_sync.readtxt',
+            None,
+            self.set_always_enable_sync_readtxt,
+        )
```

Comment:
In the end this is very similar to just overriding `Sysctl`'s attributes, but in an imperative and not declarative way. You could do something like:

```python
class ConnectionsManagerSysctl(Sysctl):
    commands: Dict[str, Tuple[Getter, Setter] = {
        'max_enabled_sync': (get_max_enabled_sync, set_max_enabled_sync),
        'rate_limit.global.send_tips': (get_global_send_tips_rate_limit, set_global_send_tips_rate_limit)
        ...
    }
```

You could also combine this with the `SysctlCommand` named tuple suggested above.

Reply:
I tested a few different syntaxes but I feel the current one is the easiest to read. I kept the `register()` method because we might refactor it to something else in the near future.

Reply:
Ok, I'm fine going how it is today